internal/scheduler/nodeiteration.go

```diff
@@ -0,0 +1,207 @@
+package scheduler
+
+import (
+	"container/heap"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/hashicorp/go-memdb"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	"github.com/G-Research/armada/internal/scheduler/schedulerobjects"
+)
+
+// NodeTypesResourceIterator extends NodeTypeResourceIterator to iterate over nodes of several node types.
+// Nodes are returned in sorted order, going from least to most of the specified resource available.
+type NodeTypesResourceIterator struct {
+	priority int32
+	resource string
+	pq       NodeTypesResourceIteratorPQ
+}
+
+func NewNodeTypesResourceIterator(txn *memdb.Txn, resource string, priority int32, nodeTypes []*schedulerobjects.NodeType, resourceQuantity resource.Quantity) (*NodeTypesResourceIterator, error) {
+	pq := make(NodeTypesResourceIteratorPQ, 0, len(nodeTypes))
+	for _, nodeType := range nodeTypes {
+		it, err := NewNodeTypeResourceIterator(txn, resource, priority, nodeType, resourceQuantity)
+		if err != nil {
+			return nil, err
+		}
+		nodeItem := it.NextNodeItem()
+		if nodeItem == nil {
+			continue // No nodes with sufficient resources for this NodeType.
+		}
+		heap.Push(&pq, &NodeTypesResourceIteratorItem{
+			value:    nodeItem,
+			it:       it,
+			priority: nodeItem.AvailableQuantityByPriorityAndResource(priority, resource),
+		})
+	}
+	return &NodeTypesResourceIterator{
+		priority: priority,
+		resource: resource,
+		pq:       pq,
+	}, nil
+}
+
+func (it *NodeTypesResourceIterator) WatchCh() <-chan struct{} {
```

Comment:
Is this needed to conform to some interface?  If so le's doc what the interface is and why it's ok not to implement it.

Reply:
Agreed. Let me to that.